policies/recipes/steps.go

```diff
@@ -90,8 +100,335 @@ func parsePermissions(s string) (os.FileMode, error) {
 }
 
 // StepArchiveExtraction builds the command for a ArchiveExtraction step
-func StepArchiveExtraction(step *osconfigpb.SoftwareRecipe_Step_ArchiveExtraction, artifacts map[string]string) error {
-	fmt.Println("StepArchiveExtraction")
+func StepArchiveExtraction(step *osconfigpb.SoftwareRecipe_Step_ArchiveExtraction, artifacts map[string]string, stepDir string) error {
+	filename, ok := artifacts[step.ArchiveExtraction.GetArtifactId()]
+	if !ok {
+		return fmt.Errorf("%q not found in artifact map", step.ArchiveExtraction.GetArtifactId())
+	}
+	switch step.ArchiveExtraction.GetType() {
+	case osconfigpb.SoftwareRecipe_Step_ExtractArchive_ZIP:
+		return extractZip(filename, step.ArchiveExtraction.Destination)
+	case osconfigpb.SoftwareRecipe_Step_ExtractArchive_TAR:
+		return extractTar(filename, step.ArchiveExtraction.Destination)
+	case osconfigpb.SoftwareRecipe_Step_ExtractArchive_TAR_GZIP:
+		compressed, err := os.Open(filename)
+		if err != nil {
+			return err
+		}
+		defer compressed.Close()
+		reader, err := gzip.NewReader(compressed)
+		if err != nil {
+			return err
+		}
+		defer reader.Close()
+		decompressed, err := ioutil.TempFile(stepDir, "archive-*.tar")
```

Comment:
why this name? i don't think go is going to expand a shell glob. not sure what it would expand to if it did

Reply:
It's a format particular to ioutil.Tempfile. It replaces the * with a random number, making sure that the number doesn't collide with existing files. This should be the only file in the directory so I'll switch to the os.create method.

Reply:
I had no idea about that tempfile feature, thx for explaining